### PR TITLE
Flip infacing normals

### PR DESCRIPTION
**Pull request by raheelx**
Hi @JoshEngebretson :)

This PR adds the ability to add import settings when importing models. It was needed because of the problem when adding planar models to the scene. 

How it Works:
1. Create an Import.json file in the Settings Folder of your project.
2. The Import.json allows you to set each of the postprocess flags to true or false. This is what it looks like:

```
{
  "desktop": {

    "aiFlagsDefault" : {
      "convertToLeftHanded" : true,
      "joinIdenticalVertices" : true,
      "triangulate" : true, 
      "genSmoothNormals" : true,
      "limitBoneWeights" : true,
      "improveCacheLocality" : true,
      "fixInfacingNormals" : false,
      "findInvalidData" : true,
      "genUVCoords" : true,
      "findInstances" : true,
      "optimizeMeshes" : true
    }
  }

}
```

NB: If there isn't an Import.json in the Settings folder of the game, the engine will set the flags to its default behaviour (where everything is true).

Is this okay? Please let me know if there is anything you'd like to be different. 

-Raheel

### CODE REVIEW

**Comment by raheelx** [comment]
Please don't land this just yet, we've found an issue.

**Comment by raheelx** [comment]
The issue was solved, so it can be tested now :+1:

**Comment by JoshEngebretson** [comment]
Cool, having these exposed is great and a good bootstrap to expose the functionality per project.  We do need better per model import settings.  Assimp has some FBX limitations, may have to look into Autodesk's FBX SDK though thinking the license may be an issue there.

File: Source/Atomic/Engine/EngineConfig.h [hunk 1]
```diff
@@ -24,43 +24,37 @@
...
+#include "../Resource/Configuration.h"
 
 namespace Atomic
 {
 
 class Context;
 
 /// Atomic engine configuration
-class EngineConfig
+class EngineConfig :
```
Comment (JoshEngebretson) [inline Source/Atomic/Engine/EngineConfig.h]: Good introduction of inheritance

File: Source/Atomic/Resource/ImportConfig.cpp [hunk 2]
```diff
@@ -0,0 +1,89 @@
+//
```
Comment (JoshEngebretson) [inline Source/Atomic/Resource/ImportConfig.cpp]: This file should be in Source/ToolCore/Import most likely

File: Source/Atomic/Resource/ImportConfig.cpp [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+            valueMap_["aiProcess_FixInfacingNormals"] = GetBoolValue(jvalue, true);
+        else if (key == "findInvalidData")
+            valueMap_["aiProcess_FindInvalidData"] = GetBoolValue(jvalue, true);
+        else if (key == "genUVCoords")
+            valueMap_["aiProcess_GenUVCoords"] = GetBoolValue(jvalue, true);
+        else if (key == "findInstances")
+            valueMap_["aiProcess_FindInstances"] = GetBoolValue(jvalue, true);
+        else if (key == "optimizeMeshes")
+            valueMap_["aiProcess_OptimizeMeshes"] = GetBoolValue(jvalue, true);
+    }
```
Comment (JoshEngebretson) [inline Source/Atomic/Resource/ImportConfig.cpp]: You've GOT to have options! :+1:

File: Source/Atomic/Resource/ImportConfig.cpp [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
+    return true;
+}
+
+bool ImportConfig::LoadDesktopConfig(JSONValue root)
+{
+    const JSONValue& jdesktop = root["desktop"];
```
Comment (JoshEngebretson) [inline Source/Atomic/Resource/ImportConfig.cpp]: I wonder if we should have quality levels here, which I am thinking "desktop" implies high quality settings?

File: Source/ToolCore/Import/OpenAssetImporter.cpp [hunk 5]
```diff
@@ -860,6 +866,70 @@ void OpenAssetImporter::CollectAnimations(OutModel* model)
...
 }
 
+void OpenAssetImporter::ApplyFlag(int flag, bool active)
+{
+    aiFlagsDefault_ &= ~flag;
+    if (active)
+        aiFlagsDefault_ |= flag;
+}
+
+void OpenAssetImporter::SetOveriddenFlags(VariantMap aiFlagParameters)
```
Comment (JoshEngebretson) [inline Source/ToolCore/Import/OpenAssetImporter.cpp]: Can this be (const VariantMap& aiFlagParameters, otherwise making a copy when passing to the method

**Comment by JoshEngebretson** [comment]
A nit and a couple questions, otherwise looking good :)

File: Source/Atomic/Resource/ImportConfig.cpp (hunk 4 above)
Comment (mattbenic) [inline Source/Atomic/Resource/ImportConfig.cpp]: Do you mean like Unity's high/medium/low levels? If so would that be under each pleatform? So you'd have the reading of each set of settings under "Desktop" for example, then optional overrides for "Desktop-High", "Desktop-Medium", "Desktop-Low"? And down the line of course, not holding up this PR, right?

Comment (JoshEngebretson) [inline Source/Atomic/Resource/ImportConfig.cpp]: The only thing I can really think of on the OpenAssetImporter is limiting bone weights or something based on quality settings and that is a stretch, so definitely not holding up PR on it...

**Comment by JoshEngebretson** [comment]
Looks good, this will probably get refactored a bit in the future when exposing to UI.  Thanks, LANDING!

**Pull request merged**